libraries/stdlib/samples/test/samples/properties/delegates.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package samples.delegates
+
+import samples.*
+import kotlin.test.*
+
+class Delegates {
+    @Sample
+    fun vetoableDelegate() {
+        var max: Int by Delegates.vetoable(0) { property, oldValue, newValue ->
+            newValue > oldValue
```

Comment:
Another common use case of `vetoable` is to throw an exception before the new value is assigned if it is invalid. It could be illustrated in a separate sample.

Reply:
Added this in throwVetoableDelegate() sample.